### PR TITLE
Reduce the amount of EGL buffer swapping the ML port is doing

### PR DESCRIPTION
**Pull request by asajeffrey**
At the moment, we're swapping the EGL buffers every heartbeat, which produces a lot of flickering. We should only swap the buffers when needed.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because we have no way to test on ML

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22044)

### CODE REVIEW

File: ports/libmlservo/src/lib.rs [hunk 1]
```diff
@@ -110,28 +119,23 @@ pub unsafe extern "C" fn discard_servo(servo: ServoInstance) {
...
-}
-
 impl WindowMethods for WindowInstance {
     fn present(&self) {
+        SwapBuffers(self.disp, self.surf);
     }
 
     fn prepare_for_composite(&self, _w: Length<u32, DevicePixel>, _h: Length<u32, DevicePixel>) -> bool {
+        MakeCurrent(self.disp, self.surf, self.surf, self.ctxt);
+        self.gl.viewport(0, 0, self.width as i32, self.height as i32);
```
Comment (jdm) [inline ports/libmlservo/src/lib.rs]: Shouldn't this be using the w and h arguments instead?

**Review by jdm — CHANGES_REQUESTED** [review]

File: ports/libmlservo/src/lib.rs (hunk 1 above)
Comment (asajeffrey) [reply ports/libmlservo/src/lib.rs]: Don't know. On sample runs they agree, except on the very first call where `_w` and `_h` are zero. I think the GL viewport has to agree with the viewport reported by `get_coordinates()` which uses `self.x` and `self.y`.

Comment (paulrouget) [reply ports/libmlservo/src/lib.rs]: The very first call is just to make sure we call make_current early: https://github.com/servo/servo/pull/16903

**Comment by jdm** [comment]
@bors-servo retry
* #22050

**Pull request merged**